lib/merkle-tree.js

```diff
@@ -1224,7 +1227,25 @@ function getStoredNode (storage, index, cache, error) {
       }
 
       const node = { index, size, hash }
-      if (cache !== null) cache.set(index, node)
+
+      if (cache !== null) {
```

Comment:
Note: the logic could be simplified if we always copy the node's buffers (not only when cache is enabled). I think that would make sense too